state/application.go

```diff
@@ -1005,6 +1005,108 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// UpdateApplicationSeries updates the series for the Application.
+func (a *Application) UpdateApplicationSeries(series string, force bool) (err error) {
+	acopy := &Application{a.st, a.doc}
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		a := acopy
+		if attempt > 0 {
+			// If we've tried once already and failed, re-evaluate the criteria.
```

Comment:
On attempt > 0, we need to check that someone hasn't already updated the series to what we want; in that case we return ErrNoOperations. And there needs to be a test for that.
And we need to check for that in business logic outside the buildTxn func